runtime/src/comm/ugni/comm-ugni.c

```diff
@@ -5043,8 +5045,9 @@ void strd_nb_helper(chpl_comm_nb_handle_t (*xferFn)(void*, int32_t, void*,
 
   if (currHandles >= strd_maxHandles) {
     // reached max in flight -- retire some to make room
+    int spins = 0;
```

Comment:
This spin-wait loop is waiting for a network response which will usually take somewhat less than 1 microsecond to arrive, and never more than just under 2 microseconds.  If there's nothing else going on (that is, there is no other task for this thread to switch to, so it will come back to this one right away), how many QSBR checkpoints will happen in that time?  If more than one, is there any functional reason that we need more than one, or would one be enough?  I ask because if the QSBR checkpoint takes any significant time it could reduce the throughput of strided remote transfers here if we did more than one, but doing one on the first yield call in the spin-wait probably wouldn't have any effect.

Reply:
Perhaps I should profile calls to all QSBR functions, checkpoint included, and add that to the documentation. As well it can be used to make more informed decisions. In any event, you're right that in this particular case it is best to just call it at most once.